openerp/addons/base/res/res_users.py

```diff
@@ -163,13 +163,10 @@ def _is_share(self, cr, uid, ids, name, args, context=None):
         return res
 
     def _get_users_from_group(self, cr, uid, ids, context=None):
-        result = set()
-        groups = self.pool['res.groups'].browse(cr, uid, ids, context=context)
-        # Clear cache to avoid perf degradation on databases with thousands of users
-        groups.invalidate_cache()
-        for group in groups:
-            result.update(user.id for user in group.users)
-        return list(result)
+        group_user = self.pool['ir.model.data'].xmlid_to_object(cr, SUPERUSER_ID, 'base.group_user', context=context)
+        if group_user and group_user.id in ids:
+            return group_user.users.ids
+        return []
```

Comment:
It may be generic but being a store trigger, it's placed on the wrong model.
You have to call it this way
```python
env['res.users']._get_users_from_group(some_group_ids)
```

which is not really intuitive. And this is can be trivially written using `mapped`

Anyway, I can keep the old function if you want.